scripting/src/main/java/org/infinispan/scripting/impl/ScriptMetadata.java

```diff
@@ -148,31 +217,55 @@ public Integer getId() {
 
       @Override
       public Set<Class<? extends ScriptMetadata>> getTypeClasses() {
-         return Util.<Class<? extends ScriptMetadata>>asSet(ScriptMetadata.class);
+         return Util.<Class<? extends ScriptMetadata>> asSet(ScriptMetadata.class);
       }
 
       @Override
       public void writeObject(ObjectOutput output, ScriptMetadata object) throws IOException {
-         output.writeInt(object.properties.size());
-         for(Entry<MetadataProperties, String> e : object.properties.entrySet()) {
-            output.writeUTF(e.getKey().name());
-            output.writeUTF(e.getValue());
-         }
+         output.writeUTF(object.name);
+         output.writeUTF(object.extension);
+         output.writeUTF(object.mode.name());
+         writeSet(output, object.parameters);
+         output.writeObject(object.language);
+         output.writeObject( object.role);
+         output.writeObject(object.reducer);
+         output.writeObject(object.collator);
+         output.writeObject(object.combiner);
       }
 
       @Override
       public ScriptMetadata readObject(ObjectInput input) throws IOException, ClassNotFoundException {
          Builder builder = new ScriptMetadata.Builder();
-         int size = input.readInt();
-         for(int i = 0; i < size; i++) {
-            String key = input.readUTF();
-            String value = input.readUTF();
-            builder.property(MetadataProperties.valueOf(key), value);
-         }
+         builder.name(input.readUTF()).extension(input.readUTF()).mode(ExecutionMode.valueOf(input.readUTF()));
+         builder.parameters(readSet(input));
+
+         ((Optional<String>)input.readObject()).ifPresent(s -> builder.language(s));
+         ((Optional<String>)input.readObject()).ifPresent(s -> builder.role(s));
+         ((Optional<String>)input.readObject()).ifPresent(s -> builder.reducer(s));
+         ((Optional<String>)input.readObject()).ifPresent(s -> builder.collator(s));
+         ((Optional<String>)input.readObject()).ifPresent(s -> builder.combiner(s));
          return builder.build();
```

Comment:
This makes it more readable, otherwise you'd have a bunch of (..., input.readObject(), ...) and you wouldn't know which one is which :)